### ISSUE TITLE
Nemo toolkit installation failed in google colab

### ISSUE THREAD

**Issue by dawnuuuu** [NVIDIA/NeMo#7109]
I kept on trying to install nemo toolkit but I keep on getting this error:
```jupyter-notebook
Collecting wget
  Downloading wget-3.2.zip (10 kB)
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: wget
  Building wheel for wget (setup.py) ... done
  Created wheel for wget: filename=wget-3.2-py3-none-any.whl size=9657 sha256=9d056575abefb14615c70a2e0e473b362f9821047bd7a3da45bbb2d36aab773f
  Stored in directory: /root/.cache/pip/wheels/8b/f1/7f/5c94f0a7a505ca1c81cd1d9208ae2064675d97582078e6c769
Successfully built wget
Installing collected packages: wget
Successfully installed wget-3.2
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
libsndfile1 is already the newest version (1.0.31-2build1).
ffmpeg is already the newest version (7:4.4.2-0ubuntu0.22.04.1).
The following additional packages will be installed:
  libid3tag0 libmad0 libopencore-amrnb0 libopencore-amrwb0 libsox-fmt-alsa
  libsox-fmt-base libsox3 libwavpack1
Suggested packages:
  libsox-fmt-all
The following NEW packages will be installed:
  libid3tag0 libmad0 libopencore-amrnb0 libopencore-amrwb0 libsox-fmt-alsa
  libsox-fmt-base libsox-fmt-mp3 libsox3 libwavpack1 sox
0 upgraded, 10 newly installed, 0 to remove and 8 not upgraded.
Need to get 729 kB of archives.
After this operation, 2,146 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libid3tag0 amd64 0.15.1b-14 [31.3 kB]
Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libmad0 amd64 0.15.1b-10ubuntu1 [63.1 kB]
Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libopencore-amrnb0 amd64 0.1.5-1 [94.8 kB]
Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libopencore-amrwb0 amd64 0.1.5-1 [49.1 kB]
Get:5 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 libsox3 amd64 14.4.2+git20190427-2+deb11u2build0.22.04.1 [240 kB]
Get:6 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 libsox-fmt-alsa amd64 14.4.2+git20190427-2+deb11u2build0.22.04.1 [11.2 kB]
Get:7 http://archive.ubuntu.com/ubuntu jammy/main amd64 libwavpack1 amd64 5.4.0-1build2 [83.7 kB]
Get:8 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 libsox-fmt-base amd64 14.4.2+git20190427-2+deb11u2build0.22.04.1 [33.7 kB]
Get:9 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 libsox-fmt-mp3 amd64 14.4.2+git20190427-2+deb11u2build0.22.04.1 [17.4 kB]
Get:10 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 sox amd64 14.4.2+git20190427-2+deb11u2build0.22.04.1 [104 kB]
Fetched 729 kB in 2s (368 kB/s)
Selecting previously unselected package libid3tag0:amd64.
(Reading database ... 129824 files and directories currently installed.)
Preparing to unpack .../0-libid3tag0_0.15.1b-14_amd64.deb ...
Unpacking libid3tag0:amd64 (0.15.1b-14) ...
Selecting previously unselected package libmad0:amd64.
Preparing to unpack .../1-libmad0_0.15.1b-10ubuntu1_amd64.deb ...
Unpacking libmad0:amd64 (0.15.1b-10ubuntu1) ...
Selecting previously unselected package libopencore-amrnb0:amd64.
Preparing to unpack .../2-libopencore-amrnb0_0.1.5-1_amd64.deb ...
Unpacking libopencore-amrnb0:amd64 (0.1.5-1) ...
Selecting previously unselected package libopencore-amrwb0:amd64.
Preparing to unpack .../3-libopencore-amrwb0_0.1.5-1_amd64.deb ...
Unpacking libopencore-amrwb0:amd64 (0.1.5-1) ...
Selecting previously unselected package libsox3:amd64.
Preparing to unpack .../4-libsox3_14.4.2+git20190427-2+deb11u2build0.22.04.1_amd64.deb ...
Unpacking libsox3:amd64 (14.4.2+git20190427-2+deb11u2build0.22.04.1) ...
Selecting previously unselected package libsox-fmt-alsa:amd64.
Preparing to unpack .../5-libsox-fmt-alsa_14.4.2+git20190427-2+deb11u2build0.22.04.1_amd64.deb ...
Unpacking libsox-fmt-alsa:amd64 (14.4.2+git20190427-2+deb11u2build0.22.04.1) ...
Selecting previously unselected package libwavpack1:amd64.
Preparing to unpack .../6-libwavpack1_5.4.0-1build2_amd64.deb ...
Unpacking libwavpack1:amd64 (5.4.0-1build2) ...
Selecting previously unselected package libsox-fmt-base:amd64.
Preparing to unpack .../7-libsox-fmt-base_14.4.2+git20190427-2+deb11u2build0.22.04.1_amd64.deb ...
Unpacking libsox-fmt-base:amd64 (14.4.2+git20190427-2+deb11u2build0.22.04.1) ...
Selecting previously unselected package libsox-fmt-mp3:amd64.
Preparing to unpack .../8-libsox-fmt-mp3_14.4.2+git20190427-2+deb11u2build0.22.04.1_amd64.deb ...
Unpacking libsox-fmt-mp3:amd64 (14.4.2+git20190427-2+deb11u2build0.22.04.1) ...
Selecting previously unselected package sox.
Preparing to unpack .../9-sox_14.4.2+git20190427-2+deb11u2build0.22.04.1_amd64.deb ...
Unpacking sox (14.4.2+git20190427-2+deb11u2build0.22.04.1) ...
Setting up libsox3:amd64 (14.4.2+git20190427-2+deb11u2build0.22.04.1) ...
Setting up libid3tag0:amd64 (0.15.1b-14) ...
Setting up libopencore-amrwb0:amd64 (0.1.5-1) ...
Setting up libsox-fmt-alsa:amd64 (14.4.2+git20190427-2+deb11u2build0.22.04.1) ...
Setting up libmad0:amd64 (0.15.1b-10ubuntu1) ...
Setting up libwavpack1:amd64 (5.4.0-1build2) ...
Setting up libopencore-amrnb0:amd64 (0.1.5-1) ...
Setting up libsox-fmt-base:amd64 (14.4.2+git20190427-2+deb11u2build0.22.04.1) ...
Setting up libsox-fmt-mp3:amd64 (14.4.2+git20190427-2+deb11u2build0.22.04.1) ...
Setting up sox (14.4.2+git20190427-2+deb11u2build0.22.04.1) ...
Processing triggers for man-db (2.10.2-1) ...
Processing triggers for libc-bin (2.35-0ubuntu3.1) ...
Requirement already satisfied: text-unidecode in /usr/local/lib/python3.10/dist-packages (1.3)
DEPRECATION: git+https://github.com/NVIDIA/NeMo.git@r1.19.1#egg=nemo_toolkit[all] contains an egg fragment with a non-PEP 508 name pip 25.0 will enforce this behaviour change. A possible replacement is to use the req @ url syntax, and remove the egg fragment. Discussion can be found at https://github.com/pypa/pip/issues/11617
Collecting nemo_toolkit[all]
  Cloning https://github.com/NVIDIA/NeMo.git (to revision r1.19.1) to /tmp/pip-install-z_sduiuz/nemo-toolkit_20856122c60348d8bf748ed8dbe6478b
  Running command git clone --filter=blob:none --quiet https://github.com/NVIDIA/NeMo.git /tmp/pip-install-z_sduiuz/nemo-toolkit_20856122c60348d8bf748ed8dbe6478b
  Running command git checkout -b r1.19.1 --track origin/r1.19.1
  Switched to a new branch 'r1.19.1'
  Branch 'r1.19.1' set up to track remote branch 'r1.19.1' from 'origin'.
  Resolved https://github.com/NVIDIA/NeMo.git to commit 55f8ce5a32ee2a2ec2c50c6df3eb8af585ea2fae
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting huggingface-hub (from nemo_toolkit[all])
  Downloading huggingface_hub-0.16.4-py3-none-any.whl (268 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 268.8/268.8 kB 4.7 MB/s eta 0:00:00
Requirement already satisfied: numba in /usr/local/lib/python3.10/dist-packages (from nemo_toolkit[all]) (0.56.4)
Requirement already satisfied: numpy>=1.22 in /usr/local/lib/python3.10/dist-packages (from nemo_toolkit[all]) (1.22.4)
Collecting onnx>=1.7.0 (from nemo_toolkit[all])
  Downloading onnx-1.14.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (14.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 14.6/14.6 MB 85.3 MB/s eta 0:00:00
Requirement already satisfied: python-dateutil in /usr/local/lib/python3.10/dist-packages (from nemo_toolkit[all]) (2.8.2)
Collecting ruamel.yaml (from nemo_toolkit[all])
  Downloading ruamel.yaml-0.17.32-py3-none-any.whl (112 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 112.2/112.2 kB 18.6 MB/s eta 0:00:00
Requirement already satisfied: scikit-learn in /usr/local/lib/python3.10/dist-packages (from nemo_toolkit[all]) (1.2.2)
Collecting setuptools==65.5.1 (from nemo_toolkit[all])
  Downloading setuptools-65.5.1-py3-none-any.whl (1.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 24.9 MB/s eta 0:00:00
Requirement already satisfied: tensorboard in /usr/local/lib/python3.10/dist-packages (from nemo_toolkit[all]) (2.12.3)
Requirement already satisfied: text-unidecode in /usr/local/lib/python3.10/dist-packages (from nemo_toolkit[all]) (1.3)
Requirement already satisfied: torch in /usr/local/lib/python3.10/dist-packages (from nemo_toolkit[all]) (2.0.1+cu118)
Requirement already satisfied: tqdm>=4.41.0 in /usr/local/lib/python3.10/dist-packages (from nemo_toolkit[all]) (4.65.0)
Requirement already satisfied: wget in /usr/local/lib/python3.10/dist-packages (from nemo_toolkit[all]) (3.2)
Requirement already satisfied: wrapt in /usr/local/lib/python3.10/dist-packages (from nemo_toolkit[all]) (1.14.1)
Collecting black==19.10b0 (from nemo_toolkit[all])
  Downloading black-19.10b0-py36-none-any.whl (97 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.5/97.5 kB 14.7 MB/s eta 0:00:00
Collecting click==8.0.2 (from nemo_toolkit[all])
  Downloading click-8.0.2-py3-none-any.whl (97 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.6/97.6 kB 16.4 MB/s eta 0:00:00
Collecting isort<6.0.0,>5.1.0 (from nemo_toolkit[all])
  Downloading isort-5.12.0-py3-none-any.whl (91 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 91.2/91.2 kB 9.8 MB/s eta 0:00:00
Collecting parameterized (from nemo_toolkit[all])
  Downloading parameterized-0.9.0-py2.py3-none-any.whl (20 kB)
Requirement already satisfied: pytest in /usr/local/lib/python3.10/dist-packages (from nemo_toolkit[all]) (7.2.2)
Collecting pytest-runner (from nemo_toolkit[all])
  Downloading pytest_runner-6.0.0-py3-none-any.whl (7.2 kB)
Requirement already satisfied: sphinx in /usr/local/lib/python3.10/dist-packages (from nemo_toolkit[all]) (3.5.4)
Collecting sphinxcontrib-bibtex (from nemo_toolkit[all])
  Downloading sphinxcontrib_bibtex-2.5.0-py3-none-any.whl (39 kB)
Collecting wandb (from nemo_toolkit[all])
  Downloading wandb-0.15.7-py3-none-any.whl (2.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 76.5 MB/s eta 0:00:00
Collecting hydra-core<1.3,>=1.2.0 (from nemo_toolkit[all])
  Downloading hydra_core-1.2.0-py3-none-any.whl (151 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 151.1/151.1 kB 23.8 MB/s eta 0:00:00
Collecting omegaconf<2.3,>=2.2 (from nemo_toolkit[all])
  Downloading omegaconf-2.2.3-py3-none-any.whl (79 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 79.3/79.3 kB 7.9 MB/s eta 0:00:00
Collecting pytorch-lightning<=1.9.4,>=1.9.0 (from nemo_toolkit[all])
  Downloading pytorch_lightning-1.9.4-py3-none-any.whl (827 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 827.8/827.8 kB 80.1 MB/s eta 0:00:00
Collecting pyyaml<6 (from nemo_toolkit[all])
  Downloading PyYAML-5.4.1.tar.gz (175 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 175.1/175.1 kB 25.8 MB/s eta 0:00:00
  Installing build dependencies ... done
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  Getting requirements to build wheel ... error
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.
``` 
**Environment details**

- Python version 3.10.6

**Comment by titu1994** [comment]
This will be fixed in NeMo 1.20

**Comment by ARAnsar** [comment]
> This will be fixed in NeMo 1.20

Do you know when this version will be released?

**Comment by dawnuuuu** [comment]
> > This will be fixed in NeMo 1.20
> 
> Do you know when this version will be released?

No idea, but I made it work by installing directly from the git repository using this command:
!pip install git+https://github.com/NVIDIA/NeMo.git@main#egg=nemo_toolkit[all]

**Comment by ARAnsar** [comment]
> > > This will be fixed in NeMo 1.20
> > 
> > 
> > Do you know when this version will be released?
> 
> No idea, but I made it work by installing directly from the git repository using this command: !pip install git+[https://github.com/NVIDIA/NeMo.git@main#egg=nemo_toolkit[all]](https://github.com/NVIDIA/NeMo.git@main#egg=nemo_toolkit%5Ball%5D)
thanks✌️
